app/models/badges/new_years_resolution_badge.rb

```diff
@@ -0,0 +1,18 @@
+module Badges
+  class NewYearsResolutionBadge < Badge
+    seed "New Years Resolution",
+      :rare,
+      'new-years-resolution',
+      'Awarded for submitting a solution on January 1st'
+
+    def award_to?(user)
+      user.solutions.
+        where('DAYOFYEAR(solutions.created_at) = 1').
+        exists?
```

Comment:
As we don't yet store the users local time (I think), this does mean that the current implementation will be incorrect for users in non-UTC timezones. I'm not yet sure what we should be doing about that.